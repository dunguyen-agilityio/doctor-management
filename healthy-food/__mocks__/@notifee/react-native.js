export default {
  requestPermission: jest.fn(() => Promise.resolve(true)),
  displayNotification: jest.fn(() => Promise.resolve()),
  onForegroundEvent: jest.fn(),
  onBackgroundEvent: jest.fn(),
  createChannel: jest.fn(() => Promise.resolve('mock-channel')),
  getInitialNotification: jest.fn(() => Promise.resolve(null)),
};
