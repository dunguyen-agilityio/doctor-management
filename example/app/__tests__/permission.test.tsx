import { render, screen } from "@testing-library/react-native";
import PermissionTest from "../permission";

describe("Permission", () => {
  test("should match the snapshot", () => {
    const component = render(<PermissionTest />);
    expect(component.getAllByRole("button")).toHaveLength(1);
    expect(component.getByText("request permissions")).toBeOnTheScreen();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
