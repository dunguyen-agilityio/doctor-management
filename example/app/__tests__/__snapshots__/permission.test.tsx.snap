// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permission should match the snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "justifyContent": "center",
      "padding": 8,
      "paddingTop": null,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "margin": 24,
        "textAlign": "center",
      }
    }
  >
    Try permissions
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        request permissions
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
